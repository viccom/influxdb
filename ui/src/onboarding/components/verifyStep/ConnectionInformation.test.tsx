// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import ConnectionInformation, {
  LoadingState,
} from 'src/onboarding/components/verifyStep/ConnectionInformation'

// Types

const setup = (override = {}) => {
  const props = {
    loading: LoadingState.NotStarted,
    bucket: 'defbuck',
    countDownSeconds: 60,
    ...override,
  }

  const wrapper = shallow(<ConnectionInformation {...props} />)

  return {wrapper}
}

describe('Onboarding.Components.ConnectionInformation', () => {
  it('renders', () => {
    const {wrapper} = setup()

    expect(wrapper.exists()).toBe(true)
  })

  it('matches snapshot if loading', () => {
    const {wrapper} = setup({loading: LoadingState.Loading})

    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot if success', () => {
    const {wrapper} = setup({loading: LoadingState.Done})

    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot if no data is found', () => {
    const {wrapper} = setup({loading: LoadingState.NotFound})

    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot if error', () => {
    const {wrapper} = setup({loading: LoadingState.Error})

    expect(wrapper).toMatchSnapshot()
  })
})
