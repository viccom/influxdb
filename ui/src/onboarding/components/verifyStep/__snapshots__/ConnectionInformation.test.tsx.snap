// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding.Components.ConnectionInformation matches snapshot if error 1`] = `
<Fragment>
  <h4
    className="wizard-step--text-state error"
  >
    Error Listening for Data
  </h4>
  <p>
    Check config and try again
  </p>
</Fragment>
`;

exports[`Onboarding.Components.ConnectionInformation matches snapshot if loading 1`] = `
<Fragment>
  <h4
    className="wizard-step--text-state loading"
  >
    Awaiting Connection...
  </h4>
  <p>
    Timeout in 60 seconds
  </p>
</Fragment>
`;

exports[`Onboarding.Components.ConnectionInformation matches snapshot if no data is found 1`] = `
<Fragment>
  <h4
    className="wizard-step--text-state error"
  >
    Data Not Found
  </h4>
  <p>
    Check config and try again
  </p>
</Fragment>
`;

exports[`Onboarding.Components.ConnectionInformation matches snapshot if success 1`] = `
<Fragment>
  <h4
    className="wizard-step--text-state success"
  >
    Connection Found!
  </h4>
  <p>
    defbuck is receiving data loud and clear!
  </p>
</Fragment>
`;
